Css centralization 1203

shaikhibrahim2000 opened the pull request:
This Pull Request centralizes and refactors styles by moving inline styles into modalStyle.tsx for better maintainability and consistency.

Key Changes:

Moved the following styles to modalStyle.tsx:

labelStyle, 
divTopBottomPadding
largeTitleStyle
smallTitleStyle
titleStyle
fullSizeContainer
 bottomSpace
 footerStyle
 wellStyle
pillContainer
pillBoxLabel
pillBox
pills 
pill

Updated imports in multiple components, including:

ChartSelectComponent.tsx, AreaUnitSelectComponent.tsx, CompareControlsComponent.tsx
BarChartComponent.tsx, LineChartComponent.tsx, PlotOED.tsx, ThreeDComponent.tsx
FooterComponent.tsx, PreferencesComponent.tsx, and others.

shaikhibrahim2000 committed:
Updated AreaUnitSelectComponent.tsx, BarControlsComponent.tsx, ChartSelectComponent.tsx, CompareControlsComponent.tsx, DateRangeComponent.tsx, GraphicRateMenuComponent.tsx, IntervalControlsComponent.tsx, MapChartSelectComponent.tsx, MeterAndGroupSelectComponent.tsx, UnitSelectComponent.tsx, and PreferencesComponent.tsx to reference labelStyle and divTopBottomPadding from modalStyle.tsx

shaikhibrahim2000 committed:
refactor: Centralize titleStyle usage across multiple components

- Imported titleStyle from modalStyle.tsx in:
  - LogMsgComponent.tsx
  - ConversionsDetailComponent.tsx
  - GroupsDetailComponent.tsx
  - MapsDetailComponent.tsx
  - MetersDetailComponent.tsx
  - UnitsDetailComponent.tsx
  - VisualUnitDetailComponent.tsx
- Improved consistency and maintainability by reusing a global style.

shaikhibrahim2000 committed:
refactor: Centralized fullSizeContainer style in modalStyle.tsx

- Replaced repeated inline styles (width: 100%, height: 100%, minHeight: 700px) with fullSizeContainer from modalStyle.tsx
- Applied refactored style in:
  - BarChartComponent.tsx
  - LineChartComponent.tsx
  - PlotOED.tsx
  - ThreeDComponent.tsx
- Improved code maintainability and consistency across components.

shaikhibrahim2000 committed:
refactor: Centralized bottomSpace, footerStyle, and wellStyle in modalStyle.tsx

- Moved bottomSpace, footerStyle, and wellStyle to modalStyle.tsx for better reusability.
- Updated the following components to use the centralized styles:
  - AreaUnitSelectComponent.tsx
  - FooterComponent.tsx
  - ThreeDPillComponent.tsx
- Improved maintainability and consistency across components.

huss commented on src/client/app/styles/modalStyle.tsx:
```diff
@@ -19,3 +19,97 @@ export const tooltipBaseStyle = {
 	display: 'inline-block',
```
Now that there are a lot of styles, what do you think about putting them all in alphabetical order so it is easier to find one? This might separate similar ones so renaming would help and standardize. For example, ``largeTitleStyle`` could become ``titleLargeStyle``.

huss commented on src/client/app/styles/modalStyle.tsx:
```diff
@@ -19,3 +19,97 @@ export const tooltipBaseStyle = {
+	margin: 0
+};
+
+export const divTopBottomPadding: React.CSSProperties = {
```
If you wanted to be more consistent you could rename so "Style" is at the end of every one. Not your original doing but would be nice.

huss commented on src/client/app/styles/modalStyle.tsx:
```diff
@@ -19,3 +19,97 @@ export const tooltipBaseStyle = {
+	border: '1px solid #e3e3e3'
+};
+
+// TODO Styling for the component, may need to be converted into .css files
```
I think this is effectively put into a CSS file so this line of comment should go.

huss committed:
clean up blank lines & multiple imports

huss committed:
Merge remote-tracking branch 'origin/development' into pr/shaikhibrahim2000/1442

huss commented:
I removed blank lines and combined some imports and pushed my changes. @shaikhibrahim2000 can review to see if it seems okay to them.

huss reviewed (changes requested):
Thanks to @shaikhibrahim2000 for working on this issue. I've made a few comments to consider. Here are a few that are more general:

- marginRight: '10px' seem to be reused. Could it go into a central CSS style?
- I merged in development since your version was moderately out of date. In particular, I noticed that the new Compare Line code has some styling that should be centralized.
- There is tooltipBaseStyle already defined centrally. (It uses ``fontSize: '60%'`` but most use 50% so maybe change to that if it works okay). It is used in a number of places, e.g., src/client/app/components/admin/users/CreateUserModalComponent.tsx, to then add in the needed info on the page. I think this should be done for all the tooltip styling to centralize the basic look.
- I searched for React.CSSProperties and found it used in a number of places. Could it be centralized for some of them too?

shaikhibrahim2000 committed:
Centralized checkbox and tooltip styles across components

- Applied centralized checkbox styling in:
  - AreaUnitSelectComponent.tsx
  - ErrorBarComponent.tsx
  - BarControlsComponent.tsx
  - PreferencesComponent.tsx (also implemented checkboxes here)

- Imported and applied tooltipBaseStyle with fontSize 50% in:
  - CreateUserModalComponent.tsx
  - UsersDetailComponent.tsx
  - MetersCSVUploadComponent.tsx
  - GroupsDetailComponent.tsx
  - MapsDetailComponent.tsx
  - MetersDetailComponent.tsx
  - UnitsDetailComponent.tsx
  - VisualUnitDetailComponent.tsx

- Updated tooltipBaseStyle import in AdminComponent.tsx and changed fontSize to 60%

- Organized modalStyle.tsx: grouped all *Style functions together and sorted CSS properties alphabetically

shaikhibrahim2000 commented:
Centralized checkbox and tooltip styles across components

- Applied centralized checkbox styling in:
  - AreaUnitSelectComponent.tsx
  - ErrorBarComponent.tsx
  - BarControlsComponent.tsx
  - PreferencesComponent.tsx (also implemented checkboxes here)

- Imported and applied tooltipBaseStyle with fontSize 50% in (all these had fontSize 60% initially):
  - CreateUserModalComponent.tsx
  - UsersDetailComponent.tsx
  - MetersCSVUploadComponent.tsx
  - GroupsDetailComponent.tsx
  - MapsDetailComponent.tsx
  - MetersDetailComponent.tsx
  - UnitsDetailComponent.tsx
  - VisualUnitDetailComponent.tsx

- Organized modalStyle.tsx: grouped all *Style functions together and sorted CSS properties alphabetically
